### PR TITLE
docs: add WASM config for Apache and Nginx

### LINKED ISSUE #1544: Recommendations to WebServer configuration for hosting WebAssembly applications
## What would you like clarification on:

Opening in response to [feedback in Gitter](https://gitter.im/uno-platform/Lobby?at=5d7bb4fda6f4896449fbb423) where Igor didn't have a smooth experience with hosting a Uno application on nginx.

Most webservers don't [yet] know how to properly serve WebAssembly applications and execution will fail with errors such as:

> "wasm streaming compile failed: TypeError: Failed to execute 'compile' on 'WebAssembly': Incorrect response MIME type. Expected 'application/wasm'."

Over at https://github.com/unoplatform/uno/blob/master/src/SolutionTemplate/UnoSolutionTemplate/Wasm/wwwroot/web.config#L17 we ship examples for hosting on IIS or via Azure. Generalized guidance is required in our documentation.

Related:

- https://github.com/WebAssembly/design/issues/981
- https://gist.github.com/WesThorburn/62ea13952749d6563ce2fb15b45f1ba8

## Concern?

- [x] Hosting Uno Applications

## For which Platform:

- [ ] iOS
- [ ] Android
- [x] WebAssembly
- [ ] Windows

### PR DESCRIPTION
**Pull request by RanadeepPolavarapu**
Resolves #1544.

Signed-off-by: Ranadeep Polavarapu <ranadeep@icloud.com>

GitHub Issue (If applicable): #1544.

## PR Type

What kind of change does this PR introduce?

- Documentation content changes (New documentation)

## What is the current behavior?

Adds documentation on how to configure a web server for reverse proxying a WASM application, specifically for Uno.

## PR Checklist

Please check if your PR fulfills the following requirements:

- [X] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [ ] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] Validated PR `Screenshots Compare Test Run` results.
- [ ] Contains **NO** breaking changes
- [ ] Associated with an issue (GitHub or internal) and uses the [automatic close keywords](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue).
- [ ] Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.

## Other information

Internal Issue (If applicable):

### CODE REVIEW

**Comment by RanadeepPolavarapu** [comment]
Just adding this snippet below to make it clear and easier for the reviewer when reviewing the *nginx* portion of the document.

image


https://github.com/unoplatform/uno/pull/4246/files#diff-131cc6567446afc10e20bc888c7f317bR58-R64.

File: doc/articles/getting-started/deployments.md [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+- [WASM Web Server Configuration](#wasm-web-server-configuration)
+  - [Nginx](#nginx)
+  - [Apache](#apache)
+
+
+Regardless of what webserver (reverse proxy) you are using, you will at core need to support the following Content (MIME) types.
+
+-   `application/wasm`
+-   `application/octet-stream`
+-   `application/octet-stream`
```
Comment (jeromelaban) [inline doc/articles/getting-started/deployments.md:12]: Duplicate ?

File: doc/articles/getting-started/deployments.md [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+-   `application/font-woff`
+-   `application/font-woff`
```
Comment (jeromelaban) [inline doc/articles/getting-started/deployments.md:14]: Duplicate ?

File: doc/articles/getting-started/deployments.md [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+  open_file_cache_errors on;
+
+  # to boost I/O on HDD we can disable access logs
+  access_log off;
+
+  # copies data between one FD and other from within the kernel
+  # faster then read() + write()
+  sendfile on;
+
+  # send headers in one peace, its better then sending them one by one
```
Comment (jeromelaban) [inline doc/articles/getting-started/deployments.md:84]:
```suggestion
  # Send headers in one piece, it's better then sending them one by one
```

File: doc/articles/getting-started/deployments.md [hunk 4]
```diff
@@ -0,0 +1,167 @@
...
+  tcp_nopush on;
+
+  # don't buffer data sent, good for small data bursts in real time
+  tcp_nodelay on;
+
+  # reduce the data that needs to be sent over network -- for testing environment
```
Comment (jeromelaban) [inline doc/articles/getting-started/deployments.md:90]: Why is this test environment only ? Wasm apps benefit a lot from compression, and even better if brotli can be enabed.

File: doc/articles/getting-started/deployments.md [hunk 5]
```diff
@@ -0,0 +1,167 @@
...
+  keepalive_requests 100000;
+
+  #########################################
+  # Just For Security Reason
+  #########################################
+
+  # Security reasons, turn off nginx versions
+  server_tokens off;
+
+  more_clear_headers Server; # Custom module: headers-more-nginx-module (https://github.com/openresty/headers-more-nginx-module)
```
Comment (jeromelaban) [inline doc/articles/getting-started/deployments.md:119]: Is this a default configuration or do users need to install something additional ?

File: doc/articles/getting-started/deployments.md [hunk 6]
```diff
@@ -0,0 +1,167 @@
+# WASM Web Server Configuration
```
Comment (jeromelaban) [inline doc/articles/getting-started/deployments.md:1]:
```suggestion
# How to host a WebAssembly App
```

File: doc/articles/getting-started/deployments.md [hunk 7]
```diff
@@ -0,0 +1,167 @@
...
+
+- [WASM Web Server Configuration](#wasm-web-server-configuration)
+  - [Nginx](#nginx)
+  - [Apache](#apache)
+
+
+Regardless of what webserver (reverse proxy) you are using, you will at core need to support the following Content (MIME) types.
```
Comment (jeromelaban) [inline doc/articles/getting-started/deployments.md:8]:
```suggestion
Regardless of the webserver (or reverse proxy) software used, the support the following Content (MIME) types are always needed:
```

File: doc/articles/getting-started/deployments.md [hunk 8]
```diff
@@ -0,0 +1,167 @@
...
+
+-   `application/wasm`
+-   `application/octet-stream`
+-   `application/octet-stream`
+-   `application/font-woff`
+-   `application/font-woff`
+
+## Nginx
+
+Below contains a extremly performance tuned version of *nginx* fitting WASM deployments. The MIME types have been extended on top of the base *nginx* MIME types with WASM support.
```
Comment (jeromelaban) [inline doc/articles/getting-started/deployments.md:18]:
```suggestion
Below contains a performance tuned version of *nginx* fitting WebAssembly deployments. The MIME types have been extended on top of the base *nginx* MIME types with WASM support.
```

**Review by jeromelaban — COMMENTED** [review]
Thanks for your contribution! Could you move the file to https://github.com/unoplatform/uno/tree/master/doc/articles, and name if "how-to-host-a-webassembly-app.md"

File: doc/articles/getting-started/deployments.md (hunk 1 above)
Comment (RanadeepPolavarapu) [reply doc/articles/getting-started/deployments.md]: Yes, I have removed the duplicate.

File: doc/articles/getting-started/deployments.md (hunk 4 above)
Comment (RanadeepPolavarapu) [reply doc/articles/getting-started/deployments.md]: Added Brotli & removed test environment comment.

File: doc/articles/getting-started/deployments.md (hunk 5 above)
Comment (RanadeepPolavarapu) [reply doc/articles/getting-started/deployments.md]: Removed this as it requires an additional module.

**Comment by RanadeepPolavarapu** [comment]
> Thanks for your contribution! Could you move the file to https://github.com/unoplatform/uno/tree/master/doc/articles, and name if "how-to-host-a-webassembly-app.md"

@jeromelaban Thank you for the feedback. I have incorporated your suggestions and also moved & renamed the document.

**Review by jeromelaban — APPROVED** [review]
Thank you for your contribution !

**Pull request merged**